modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,307 @@
+        Supported value is hs2019.
+    :param signing_algorithm: The signature algorithm, when signing HTTP requests.
+        Supported values are PKCS1v15, PSS; fips-186-3, deterministic-rfc6979.
+    :param signature_max_validity: The signature max validity,
+        expressed as a datetime.timedelta value.
+    :param signed_headers: A list of strings. Each value is the name of a HTTP header
+        that must be included in the HTTP signature calculation.
+        The two special signature headers '(request-target)' and '(created)' SHOULD be
+        included in SignedHeaders.
+        The '(created)' header expresses when the signature was created.
+        The '(request-target)' header is a concatenation of the lowercased :method, an
+        ASCII space, and the :path pseudo-headers.
+        When signed_headers is not specified, the client defaults to a single value,
+        '(created)', in the list of HTTP headers.
+        When SignedHeaders contains the 'Digest' value, the client performs the
+        following operations:
+        1. Calculate a digest of request body, as specified in RFC3230, section 4.3.2.
+        2. Set the 'Digest' header in the request body.
+        3. Include the 'Digest' header and value in the HTTP signature.
+    """
+    def __init__(self, key_id=None, private_key_path=None, signing_scheme=None,
+                 signing_algorithm=None, signature_max_validity=None, signed_headers=None):
+        """Constructor
+        """
+        self.key_id = key_id
+        """The identifier of the key used to sign HTTP requests.
+        """
+        self.private_key_path = private_key_path
+        """The path of the file containing a private key, used to sign HTTP requests.
+        """
+        self.signing_scheme = signing_scheme
+        """The signature scheme when signing HTTP requests.
+           Supported values are hs2019, rsa-sha256, rsa-sha512.
+        """
+        self.signing_algorithm = signing_algorithm
+        """The signature algorithm when signing HTTP requests.
+           For RSA keys, supported values are PKCS1v15, PSS.
+           For ECDSA keys, supported values are fips-186-3, deterministic-rfc6979.
+        """
+        if signature_max_validity is not None and signature_max_validity.total_seconds() < 0:
+            raise Exception("The signature max validity must be a positive value")
+        self.signature_max_validity = signature_max_validity
+        """The signature max validity, expressed as a datetime.timedelta value.
+        It must be a positive value.
+        """
+        if self.signature_max_validity is None and \
+                signed_headers is not None and '(expires)' in signed_headers:
+            raise Exception(
+                "Signature max validity must be set when " \
+                "'(expires)' signature parameter is specified")
+        if len(signed_headers) != len(set(signed_headers)):
```

Comment:
If signed headers is optional/can be set to None then please move these checks behind the if statement:
`if signed_headers is not None:`
We are getting a CI error from tying to run len(signed_headers) when signed_headers = None

Reply:
good point, in golang this is not needed and I keep thinking golang.